cypress/e2e/shared/notificationRules.test.ts

```diff
@@ -292,4 +302,457 @@ describe('NotificationRules', () => {
+              '20m'
+            )
+            writeMonitoringRecord(
+              {
+                check: check,
+                endp: endp,
+                rule: rule,
+                level: 'ok',
+                source: 'wumpus',
+                sent: 'true',
+                valField: {key: 'foo', val: 'bar'},
+                sourceTimestamp: calcNanoTimestamp('7m'),
+                statusTimestamp: calcNanoTimestamp('6m'),
+                message: 'This is a fake OK message',
+              },
+              '5m'
+            )
+          })
+        })
+      })
+
+      // use direct route to page
+      cy.get<Organization>('@org').then(({id}: Organization) => {
+        cy.get<NotificationRule>('@rule').then(rule => {
+          cy.fixture('routes').then(({orgs}) => {
+            cy.visit(
+              `${orgs}/${id}/alert-history?type=notifications&filter="notificationRuleID%20%3D%3D%20%22${rule.id}"`
+            )
+            cy.url().should('include', `${orgs}/${id}/alert-history`)
+          })
+        })
+      })
+    })
+
+    it('shows and filters history', () => {
+      cy.get('.event-row').should('have.length', 4)
+      cy.get('.event-row--field:nth-of-type(1)').then(timestamps => {
+        const stamps = timestamps
+          .toArray()
+          .map(n => new Date(n.innerText).getTime())
+        // assert sort desc by date
+        for (let i = 1; i < stamps.length; i++) {
+          expect(stamps[i - 1]).to.be.above(stamps[i])
+        }
+      })
+      // verify first row
+      cy.get('[class$=ScrollContainer] > div:nth-of-type(1)').within(() => {
+        cy.get('.level-table-field--ok').should('be.visible')
+        cy.get('.sent-table-field--sent').should('be.visible')
+        cy.get('.event-row--field:nth-of-type(3)').should(
+          'have.text',
+          check.name
+        )
+        cy.get('.event-row--field:nth-of-type(4)').should(
+          'have.text',
+          rule.name
+        )
+        cy.get('.event-row--field:nth-of-type(5)').should(
+          'have.text',
+          endp.name
+        )
+      })
+      // verify second row
+      cy.get('[class$=ScrollContainer] > div:nth-of-type(2)').within(() => {
+        cy.get('.level-table-field--info').should('be.visible')
+        cy.get('.sent-table-field--not-sent').should('be.visible')
+      })
+      // verify third row
+      cy.get('[class$=ScrollContainer] > div:nth-of-type(3)').within(() => {
+        cy.get('.level-table-field--crit').should('be.visible')
+        cy.get('.sent-table-field--sent').should('be.visible')
+      })
+      // verify fourth row
+      cy.get('[class$=ScrollContainer] > div:nth-of-type(4)').within(() => {
+        cy.get('.level-table-field--warn').should('be.visible')
+      })
+      // check links...
+      //    ...To Check
+      cy.getByTestID('overlay').should('not.exist')
+      cy.get(
+        '[class$=ScrollContainer] > div:nth-of-type(4) .event-row--field:nth-of-type(3) > a'
```

Comment:
this is a great example where a testID should be used. It's really hard to tell what you're trying to click on here just by reading through the test code

Reply:
Agreed.  But will need to add the data-testid attributes myself.  See response to comment on line 545.